File: spoj/01-easy/glutton.rb
seconds_per_day = 86400

tests = $stdin.gets.to_i

tests.times do
    total_donuts = 0
    settings = $stdin.gets.split(' ')
    while settings[0].to_i > 0

        glutton = $stdin.gets.to_i 
        total_donuts += seconds_per_day.to_i / glutton.to_i

        settings[0] = settings[0].to_i - 1
    end

    if total_donuts.to_i % settings[1].to_i == 0
        puts total_donuts.to_i / settings[1].to_i
    else
        puts total_donuts.to_i / settings[1].to_i + 1
    end
end

File: spoj/01-latwe/GLUTTON.py
seconds_per_day = 86400
tests = int(raw_input())

for _ in range (tests):
	total_donuts = 0
	settings = map(int, raw_input().split(' '))
	while settings[0] > 0:
		glutton = int(raw_input())
		total_donuts += seconds_per_day/glutton
		settings[0] -= 1
	
	if total_donuts % settings[1] == 0:
		print total_donuts/settings[1]
	else:
		print total_donuts/settings[1] + 1
